lib/urlwatch/jobs.py

```diff
@@ -230,3 +231,29 @@ def retrieve(self, job_state):
                 return response.content.decode('ascii', 'ignore')
 
         return response.text
+
+class FileJob(Job):
+    """Read the content of a local file"""
+
+    __kind__ = 'file'
+    __required__ = ('path',)
+
+    def get_location(self):
+        return str(self.path)
+
+    def retrieve(self, job_state):
+
+        path_string = str(self.path)
+
+        if not path_string:
+            raise ValueError('Path cannot be null or empty')
+
+        if path_string.startswith('file://'):
```

Comment:
We don't need to "send in" the data into requests. In the line before `requests.request()` is called (currently here: https://github.com/thp/urlwatch/blob/master/lib/urlwatch/jobs.py#L206), one can do the `if url.startswith('file://'):` check there and return early. That should be simple enough to not have to deal with extending/modifying requests, while still allowing normal file:// URLs transparently for local files.